src/Compiler/Opts/InlineHeuristics.idr

```diff
@@ -46,5 +46,14 @@ inlineHeuristics fn = do
         | Nothing => pure ()
     let True = inlineCDef fn cdef
         | False => pure ()
-    log "compiler.inline.heuristic" 25 $ "inlining heuristic decided to inline: " ++ show fn
-    unless (NoInline `elem` gdef.flags) $ setFlag EmptyFC (Resolved fnIdx) Inline
+    -- we can't inline non-public defs (unless they are marked as Inline by developer)
```

Comment:
I'll rework this comment to be clearer -- the "unless marked Inline" does not com into play here since this code only handles inlining by heuristic; marking something inline is explicit so it'll already be marked for inlining before getting to this code. Anyway, you've made me realize this comment could be better.